torch/csrc/autograd/functions/accumulate_grad.cpp

```diff
@@ -13,12 +13,13 @@ using at::Tensor;
 
 namespace torch { namespace autograd {
 
-// AccumulateGrad sets sequence_nr to the max value so it's always called 
+// AccumulateGrad sets sequence_nr to the max value so it's always called
 // ASAP during backwards.
 AccumulateGrad::AccumulateGrad(Variable variable_)
-    : Function(/*num_inputs=*/1
-              , /*sequence_nr=*/UINT64_MAX)
-              , variable(std::move(variable_)) {}
+    : Function(/*sequence_nr=*/UINT64_MAX)
+    , variable(std::move(variable_)) {
+  add_input_metadata(variable.type(), variable.sizes());
```

Comment:
Is this correct? `AccumulateGrad` shouldn't return anything, so it doesn't have any inputs

Reply:
The input to AccumulateGrad is the gradient. It has no outputs. (The previous code had `AccumulateGrad(...)  : Function(/*num_inputs=*/1, ...)`

Reply:
Ohhh right. `input_metadata` is for forward outputs (i.e. it's written from the perspective of the backward pass). Never mind the comment